Add reolink documentation (#25306)
home-assistant/home-assistant.github.io: source/_integrations/reolink.markdown

---
title: Reolink NVR/camera
description: Instructions on how to integrate Reolink devices (NVR/cameras) into Home Assistant.
ha_category:
  - Camera
ha_iot_class: Local Polling

> ha_release: 2022.12

Reviewer: Version needs to be updated as well

Author: done

ha_domain: reolink
ha_codeowners:
  - '@starkillerOG'
ha_config_flow: true
ha_platforms:
  - Camera
ha_integration_type: integration
---

The integration allows you to control [Reolink](https://reolink.com/) NVRs or cameras.

{% include integrations/config_flow.md %}

## Known reolink firmware bugs

<div class='note warning'>
There is a BUG in Reolink NVR firmware: it only sends ONVIF event-notifications if motion happened on the camera connected to its very first (index "0") channel.
Reolink is aware of this and are working on fixing the problem.
</div>

<div class='note warning'>
Reolink doorbell presses only generate ONVIF event-notifications when the doorbell is directly connected to your network.
When connecting the Reolink doorbell to a NVR, no doorbell press events are generated.
Reolink is aware of this and a solution is already implemented in beta firmware.

> </div>

Reviewer: These type of notices tend to go stable in docs

Is there an Reolink page you can link to instead?

Author: No there is no page to link to that I am aware of, I got this information from the reolink dev team directly through private messages.
Since I am working on resolving those two bugs together with reolink, I can update the docs when it is fixed.
But I am also willing to just remove the two warnings...

Reviewer: Lets present these as limitations for now.  When Reolink updates their firmware they can be updated

Author: Alright, I have commited your suggestions to call it limitations.

| :-------------------    | :---------------------------------------------------------------------------------------------------------- |
| Protocol                | Switch between the RTMP or RTSP streaming protocol.                                                         |

## Unsupported models

The following models are not to be supported:

- Battery-powered cameras
- B800: Only with NVR
- B400: Only with NVR
- D400: Only with NVR
- Lumus

## Reducing latency and load time

To get considerebly lower lattency and faster loading, a add-on can be used to convert the RTSP stream to a WebRTC stream.
1) In the configuration options of this reolink integration, select 'RTSP' as stream protocol.
2) add https://github.com/allenporter/stream-addons to the [add-on repositories](https://my.home-assistant.io/redirect/supervisor_add_addon_repository/?repository_url=https%3A%2F%2Fgithub.com%2Fallenporter%2Fstream-addons).
3) Install en start the 'RTSPtoWeb - WebRTC' add-on.
4) The [RTSPtoWebRTC integration](https://www.home-assistant.io/integrations/rtsp_to_webrtc/) should now be automatically discovered. Navigate to the integrations dashboard in Home Assistant and install this integration.

> 5) Check in the 'development tools' -> 'status' that the camera attribute 'frontend_stream_type' now shows 'web_rtc' for the reolink cameras.

Reviewer: We shouldn't tell users to use unofficial repos in the docs

Author: I removed the instructions for reduced latency using RTSPtoWeb - WebRTC add-on.

> ## Services
> 
> The Reolink integration supports all default camera [services](https://www.home-assistant.io/integrations/camera/#services) and additionally provides the following services:
> 
> ### Service `reolink_cctv.set_sensitivity`
> 
> Set the motion detection sensitivity of the camera. Either all time schedule presets can be set at once, or a specific preset can be specified.
> 
> | Service data attribute  | Optional  | Description  |
> | :---------------------- | :-------- | :----------- |
> | `entity_id`             | no        | The camera to control.
> | `sensitivity`           | no        | The sensitivity to set, a value between 1 (low sensitivity) and 50 (high sensitivity).
> | `preset`                | yes       | The time schedule preset to set. Presets can be found in the Web UI of the camera.
> 
> ### Service `reolink_cctv.set_backlight`
> 
> Optimizing brightness and contrast levels to compensate for differences between dark and bright objects using either BLC or WDR mode.
> This may improve image clarity in high contrast situations, but it should be tested at different times of the day and night to ensure there is no negative effect.
> 
> | Service data attribute  | Optional  | Description  |
> | :---------------------- | :-------- | :----------- |
> | `entity_id`             | no        | The camera to control.
> | `mode`                  | no        | The backlight parameter supports the following values: `BACKLIGHTCONTROL`: use Backlight Control `DYNAMICRANGECONTROL`: use Dynamic Range Control `OFF`: no optimization
> 
> ### Service `reolink_cctv.set_daynight`
> 
> Set the day and night mode parameter of the camera.
> 
> | Service data attribute  | Optional  | Description  |
> | :---------------------- | :-------- | :----------- |
> | `entity_id`             | no        | The camera to control.
> | `mode`                  | no        | The day and night mode parameter supports the following values: `AUTO` Auto switch between black & white mode `COLOR` Always record videos in color mode `BLACKANDWHITE` Always record videos in black & white mode.
> 
> ### Service `reolink_cctv.ptz_control`
> 
> Control the PTZ (Pan Tilt Zoom) movement of the camera.
> 
> | Service data attribute  | Optional  | Description  |
> | :---------------------- | :-------- | :----------- |
> | `entity_id`             | no        | The camera to control.
> | `command`               | no        | The command to execute. Possibe values are: `AUTO`, `DOWN`, `FOCUSDEC`, `FOCUSINC`, `LEFT`, `LEFTDOWN`, `LEFTUP`, `RIGHT`, `RIGHTDOWN`, `RIGHTUP`, `STOP`, `TOPOS`, `UP`, `ZOOMDEC` and `ZOOMINC`.
> | `preset`                | yes       | In case of the command `TOPOS`, pass the preset ID here. The possible presets are listed as attribute on the camera.
> | `speed`                 | yes       | The speed at which the camera moves. Not applicable for the commands: `STOP` and `AUTO`.
> 
> **The camera keeps moving until the `STOP` command is passed to the service.**

Reviewer: ```suggestion
```

Dropped from the MVP

ha_category:
  - Camera
ha_iot_class: Local Polling
ha_release: 2023.1
ha_domain: reolink
ha_codeowners:
  - '@starkillerOG'
ha_config_flow: true
ha_platforms:
  - Camera
ha_integration_type: integration
---

The integration allows you to control [Reolink](https://reolink.com/) NVRs or cameras.

{% include integrations/config_flow.md %}

## Camera streams

This integration creates a few camera entities, one for each stream type with different resolutions: Main, Sub, Ext, and Images.
The Sub stream camera entity is enabled by default; the other streams are disabled by default.
The Images stream provides a sequence of image snapshots giving very low latency at the cost of a very low frame rate; this can be used when the hi-res RTMP/RTSP video stream has too much lag.

## Configuration options

In the options menu, the `Protocol` can be configured to RTSP or RTMP streaming protocol. 

## Tested models

The following models have been tested and confirmed to work:

- RLN8-410 NVR
- RLC-811A
- Reolink Video Doorbell PoE
- Reolink Video Doorbell WiFi

Battery-powered cameras are not yet supported.

## Reolink firmware limitations

> - The Reolink NVR only sends ONVIF event-notifications if motion happens on the camera connected to the first (index "0") channel.

Reviewer 2: What does that mean? (as in, why should the average joe care about this?)

Author: The motion/vehicle/pet/person/visitor binary sensors will only work for the camera connected to the first port of the NVR. The binary sensors of the other camera's will only be triggered and updated if the camera connected to port 0 sees motion.

I know, it is stupid and weird, but Reolink is working on the problem.

Now that I type this, this initial basis does not include the binary sensors yet (since I removed them), but that will be the next PR in the comming days.

> - Reolink doorbell presses only generate ONVIF event notifications when the doorbell is directly connected to your network. No doorbell press events are generated when connecting the Reolink doorbell to an NVR.

Reviewer 2: Same as above, all of a sudden there is writing about ONVIF here, why would it matter?

Author: The binary sensor 'visitor' will not work.
I will make it a bit clearer.

Reviewer 2: 👍 Thanks!

Author: I clearified it a bit, is it okay like this @frenck?

---
title: Reolink NVR/camera
description: Instructions on how to integrate Reolink devices (NVR/cameras) into Home Assistant.
ha_category:
  - Camera
ha_iot_class: Local Polling
ha_release: 2023.1
ha_domain: reolink
ha_codeowners:
  - '@starkillerOG'
ha_config_flow: true
ha_platforms:
  - Camera
ha_integration_type: integration
---

The integration allows you to control [Reolink](https://reolink.com/) NVRs or cameras.

{% include integrations/config_flow.md %}

> ## Reolink firmware limitations

Reviewer 2: There is no need to create a paragraph and put two big warning boxes in it.
Just use normal text instead.

In general, known issues and limitations should be listed at the bottom of the documentation. These are not pre-requisites.

Author: done

<div class='note warning'>
The Reolink NVR only sends ONVIF event-notifications if motion happens on the camera connected to the first (index "0") channel.
</div>

<div class='note warning'>
Reolink doorbell presses only generate ONVIF event notifications when the doorbell is directly connected to your network. No doorbell press events are generated when connecting the Reolink doorbell to an NVR.
</div>

## Camera streams

This integration creates a few camera entities, one for each stream type with different resolutions: Main, Sub, Ext, and Images.
The Sub stream camera entity is enabled by default; the other streams are disabled by default.
The Images stream provides a sequence of image snapshots giving very low latency at the cost of a very low frame rate; this can be used when the hi-res RTMP/RTSP video stream has too much lag.

## Configuration options

> In the options menu, the `Protocol` can be configured to RTSP or RTMP streaming protocol. 

Reviewer 2: If this is config entry options, please use our pre-defined snippet for that (like the include for `config_flow` above, we also have one for config entry options flows).

Author: Yes it is, I updated the docs to use {% include integrations/option_flow.md %}
and {% configuration_basic %}

## Unsupported models

> The following models are not to be supported:

Reviewer 2: Please list what is supported instead.

Author: done,I know it works with a lot more models (basically the whole reolink line-up, but I do not have a list of models reported to work by users.

---
title: Reolink NVR/camera
description: Instructions on how to integrate Reolink devices (NVR/cameras) into Home Assistant.
ha_category:
  - Camera
ha_iot_class: Local Polling
ha_release: 2023.1
ha_domain: reolink
ha_codeowners:
  - '@starkillerOG'
ha_config_flow: true
ha_platforms:
  - Camera
ha_integration_type: integration
---

The integration allows you to control [Reolink](https://reolink.com/) NVRs or cameras.

{% include integrations/config_flow.md %}

> ## Known reolink firmware bugs

Reviewer: ```suggestion
## Reolink firmware limitations
```

<div class='note warning'>

> There is a BUG in Reolink NVR firmware: it only sends ONVIF event-notifications if motion happened on the camera connected to its very first (index "0") channel.
> Reolink is aware of this and are working on fixing the problem.

Reviewer: ```suggestion
The Reolink NVR only sends ONVIF event-notifications if motion happens on the camera connected to the first (index "0") channel.
```

> There is a BUG in Reolink NVR firmware: it only sends ONVIF event-notifications if motion happened on the camera connected to its very first (index "0") channel.
> Reolink is aware of this and are working on fixing the problem.

Reviewer: ```suggestion
The Reolink NVR only sends ONVIF event notifications if motion happens on the camera connected to the first (index "0") channel.
```

</div>

<div class='note warning'>

> Reolink doorbell presses only generate ONVIF event-notifications when the doorbell is directly connected to your network.
> When connecting the Reolink doorbell to a NVR, no doorbell press events are generated.
> Reolink is aware of this and a solution is already implemented in beta firmware.

Reviewer: ```suggestion
Reolink doorbell presses only generate ONVIF event notifications when the doorbell is directly connected to your network. No doorbell press events are generated when connecting the Reolink doorbell to an NVR.
```

</div>

## Camera streams

> This integration creates a few camera entities, one for each stream type with different resolutions: Main, Sub, Ext and Images.

Reviewer: ```suggestion
This integration creates a few camera entities, one for each stream type with different resolutions: Main, Sub, Ext, and Images.
```

> The Sub stream camera entity is enabled by default, the other streams are disabled by default.

Reviewer: ```suggestion
The Sub stream camera entity is enabled by default; the other streams are disabled by default.
```

> The Images stream provides a sequence of image snapshots giving very low latency at the cost of a very low frame rate, this can be used when the hi-res RTMP/RTSP video stream are too laggy.

Reviewer: ```suggestion
The Images stream provides a sequence of image snapshots giving very low latency at the cost of a very low frame rate; this can be used when the hi-res RTMP/RTSP video stream has too much lag.
```

## Configuration options

In the options menu, the following parameters can be configured:

| Parameter               | Description                                                                                                 |
| :-------------------    | :---------------------------------------------------------------------------------------------------------- |

> | Protocol                | Switch between the RTMP or RTSP streaming protocol.                                                         |

Reviewer: A table is a bit much here for a single option.  These tend to not look so great on mobile.

Author: I removed the table in favor of text.

## Unsupported models

The following models are not to be supported:

- Battery-powered cameras
- B800: Only with NVR
- B400: Only with NVR
- D400: Only with NVR

> - Lumus

Reviewer: It would be better to list the tested models instead of the unsupported ones since usually people want to buy hardware that based on what is supported and not what they know won't work

Author: I updated it to list the models I have tested.
I am sure that it works with a lot more models, but I do not have a list of which are tested and not.
I left the note about battery-powered cameras, because they will not be supported for a while (not always available for connection).
